### ISSUE TITLE
Rubocop reports an error on multiline lambdas in scopes

### ISSUE THREAD

**Issue by k-rudy** [bbatsov/rubocop#1520]
This issue has been originally reported in #1482 but has been closed by mistake as it's still reproducible in v. 0.28.0

Hey, I have a scope defined:
```ruby
scope :some_scope, ->(param) do
  ...
end
```
Rubocop claims on `Use the 'lambda' method for multi-line lambdas`

But when I change it to:
```ruby
scope :some_scope, lambda do |param|
  ...
end
```

It claims on `Use 'lambda'/'proc' instead of a plain method call.`

So what is the correct syntax in this case?

**Comment by deivid-rodriguez** [comment]
Hi @k-rudy, I've been there too... :smile: 

The syntax RuboCop expects you to write is:

    scope :some_scope, lambda { |param|
      ...
    end

What I don't know is whether there's an option to allow the first syntax you used (which I like better than RuboCop's default).

**Comment by k-rudy** [comment]
Hey @deivid-rodriguez, thanks for the hint, it passes. But I think the provided style is a violation of [this](https://github.com/bbatsov/ruby-style-guide#single-line-blocks) rule (using `{...}` for multiline blocks), isn't it?

**Comment by deivid-rodriguez** [comment]
Yep, the problem is that the `{}` and `do...end` operators have different precedence, so 

    scope :some_scope, lambda do |param|
      ...
    end

actually means

    scope(:some_scope, lambda) do |param|
      ...
    end

It's just not equivalent...

So there's no way to be consistent here, either enforce [single-line-blocks](https://github.com/bbatsov/ruby-style-guide#single-line-blocks) or [lambda-multi-line](https://github.com/bbatsov/ruby-style-guide#lambda-multi-line), but not both.

Probably the `Style/Lambda` cop could be reworked and configurations added to enforce either 
* `->` always.
* `lambda` always.
* `->` on single line / `lambda` for multiline (current behaviour).

**Comment by k-rudy** [comment]
Ok, now I see it. Thanks  @deivid-rodriguez for detailed explanation. It all makes sense now.

**Comment by Rajasree** [comment]
Please suggest a syntax, in case if we don't have any param to pass for this multiline scope.

**Comment by jonas054** [comment]
@Rajasree Whether the lambda takes a parameter or not, the alternatives are the same. I would suggest adding a local variable:

```ruby
f = lambda do |param|
  ...
end
scope :some_scope, f
```
Maybe this looks unidiomatic, and then there's the other alternative, which I thinks is ugly:
```ruby
scope :some_scope, (lambda do |param|
  ...
end)
```

**Comment by Rajasree** [comment]
@jonas054 Thank you for your response. Its working fine.

**Comment by kuon** [comment]
As I work on swift, js and rust, I find the `->` syntax preferable, so:

```
scope :foo, -> do
end
```

Whould be prefered.

Do you have plan to allow rubocop to enforce the usage of `->` and to allow the above syntax?

**Comment by jonas054** [comment]
@kuon Adding a configuration option in `Style/Lambda` to enforce `->` for lambdas regardless of length is definitely doable. Please open a new issue to request that feature.

**Comment by bronson** [comment]
Multi-line stabby-procs seem to always work work better than lambdas, and I even think they look better.  Hoping for the `->` enforcer.  Until then, this is a quick fix:

```
Style/Lambda:
  Enabled: false
```

**Comment by ypresto** [comment]
FYI:
Enforce `->` or `lambda` regardless of line count is already implemented here: #3076

**Comment by zinkkrysty** [comment]
For people googling and ending up here, this is how you now enforce the `->` syntax, regardless of how many lines:

```yaml
Style/Lambda:
  EnforcedStyle: literal
```
